src/handlers/store-standardised-invoices/fetch-s3-textract-data.test.ts

```diff
@@ -0,0 +1,430 @@
+import { fetchS3 } from "../../shared/utils";
+import { fetchS3TextractData } from "./fetch-s3-textract-data";
+
+jest.mock("../../shared/utils");
+const mockedFetchS3 = fetchS3 as jest.Mock;
+
+describe("S3 Textract data fetcher", () => {
+  let mockedS3JsonValue: any;
+  let givenBucket: string;
+  let givenKey: string;
+
+  beforeEach(() => {
+    jest.resetAllMocks();
+
+    mockedS3JsonValue = [
+      {
+        LineItemGroups: [
+          {
+            LineItems: [
+              {
+                LineItemExpenseFields: [
+                  {
+                    Type: {
+                      Text: "mocked line item field type",
+                      Confidence: 123,
+                    },
+                    ValueDetection: {
+                      Text: "mocked field value",
+                      Confidence: 345,
+                    },
+                  },
+                ],
+              },
+            ],
+          },
+        ],
+        SummaryFields: [
+          {
+            Type: {
+              Text: "mocked summary field type",
+              Confidence: 567,
+            },
+            ValueDetection: {
+              Text: "mocked summary field value",
+              Confidence: 789,
+            },
+          },
+        ],
+      },
+    ];
+
+    mockedFetchS3.mockImplementation(() => JSON.stringify(mockedS3JsonValue));
+
+    givenBucket = "given bucket";
+    givenKey = "given key";
+  });
+
+  test("S3 Textract data fetcher with fetch error", async () => {
+    const mockedError = new Error("mocked error");
+    mockedFetchS3.mockRejectedValue(mockedError);
+
+    let resultError;
+    try {
+      await fetchS3TextractData(givenBucket, givenKey);
+    } catch (error) {
+      resultError = error;
+    }
+
+    expect(resultError).toBe(mockedError);
```

Comment:
Try using:

`await expect(fetchS3TextractData(givenBucket, givenKey)).rejects.toThrowError('mocked error');`

Makes it more concise and readable. Probably best to put the `'mocked error'` string in a constant, too.